Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/nbclient/client.py", line 1319, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
from policyengine import Simulation
import pandas as pd

sim = Simulation(
    country="uk",
    scope="macro",
)
------------------

----- stderr -----
/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm
----- stdout -----
Using Hugging Face for download.
Failed to download from Hugging Face.
Using Google Cloud Storage for download.
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mDefaultCredentialsError[0m                   Traceback (most recent call last)
Cell [0;32mIn[1], line 4[0m
[1;32m      1[0m [38;5;28;01mfrom[39;00m[38;5;250m [39m[38;5;21;01mpolicyengine[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m Simulation
[1;32m      2[0m [38;5;28;01mimport[39;00m[38;5;250m [39m[38;5;21;01mpandas[39;00m[38;5;250m [39m[38;5;28;01mas[39;00m[38;5;250m [39m[38;5;21;01mpd[39;00m
[0;32m----> 4[0m sim [38;5;241m=[39m [43mSimulation[49m[43m([49m
[1;32m      5[0m [43m    [49m[43mcountry[49m[38;5;241;43m=[39;49m[38;5;124;43m"[39;49m[38;5;124;43muk[39;49m[38;5;124;43m"[39;49m[43m,[49m
[1;32m      6[0m [43m    [49m[43mscope[49m[38;5;241;43m=[39;49m[38;5;124;43m"[39;49m[38;5;124;43mmacro[39;49m[38;5;124;43m"[39;49m[43m,[49m
[1;32m      7[0m [43m)[49m

File [0;32m/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/policyengine/simulation.py:76[0m, in [0;36mSimulation.__init__[0;34m(self, **options)[0m
[1;32m     73[0m [38;5;28;01mdef[39;00m[38;5;250m [39m[38;5;21m__init__[39m([38;5;28mself[39m, [38;5;241m*[39m[38;5;241m*[39moptions: SimulationOptions):
[1;32m     74[0m     [38;5;28mself[39m[38;5;241m.[39moptions [38;5;241m=[39m SimulationOptions([38;5;241m*[39m[38;5;241m*[39moptions)
[0;32m---> 76[0m     [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_set_data[49m[43m([49m[43m)[49m
[1;32m     77[0m     [38;5;28mself[39m[38;5;241m.[39m_initialise_simulations()
[1;32m     78[0m     [38;5;28mself[39m[38;5;241m.[39m_add_output_functions()

File [0;32m/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/policyengine/simulation.py:113[0m, in [0;36mSimulation._set_data[0;34m(self)[0m
[1;32m    111[0m [38;5;28;01mdef[39;00m[38;5;250m [39m[38;5;21m_set_data[39m([38;5;28mself[39m):
[1;32m    112[0m     [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39moptions[38;5;241m.[39mdata [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m:
[0;32m--> 113[0m         [38;5;28mself[39m[38;5;241m.[39moptions[38;5;241m.[39mdata [38;5;241m=[39m [43mget_default_dataset[49m[43m([49m
[1;32m    114[0m [43m            [49m[43mcountry[49m[38;5;241;43m=[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43moptions[49m[38;5;241;43m.[39;49m[43mcountry[49m[43m,[49m
[1;32m    115[0m [43m            [49m[43mregion[49m[38;5;241;43m=[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43moptions[49m[38;5;241;43m.[39;49m[43mregion[49m[43m,[49m
[1;32m    116[0m [43m        [49m[43m)[49m
[1;32m    118[0m     [38;5;28;01melif[39;00m [38;5;28misinstance[39m([38;5;28mself[39m[38;5;241m.[39moptions[38;5;241m.[39mdata, [38;5;28mstr[39m):
[1;32m    119[0m         filename [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39moptions[38;5;241m.[39mdata

File [0;32m/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/policyengine/constants.py:17[0m, in [0;36mget_default_dataset[0;34m(country, region)[0m
[1;32m     15[0m [38;5;28;01mdef[39;00m[38;5;250m [39m[38;5;21mget_default_dataset[39m(country: [38;5;28mstr[39m, region: [38;5;28mstr[39m):
[1;32m     16[0m     [38;5;28;01mif[39;00m country [38;5;241m==[39m [38;5;124m"[39m[38;5;124muk[39m[38;5;124m"[39m:
[0;32m---> 17[0m         data_file [38;5;241m=[39m [43mdownload[49m[43m([49m
[1;32m     18[0m [43m            [49m[43mfilepath[49m[38;5;241;43m=[39;49m[38;5;124;43m"[39;49m[38;5;124;43menhanced_frs_2022_23.h5[39;49m[38;5;124;43m"[39;49m[43m,[49m
[1;32m     19[0m [43m            [49m[43mhuggingface_repo[49m[38;5;241;43m=[39;49m[38;5;124;43m"[39;49m[38;5;124;43mpolicyengine-uk-data[39;49m[38;5;124;43m"[39;49m[43m,[49m
[1;32m     20[0m [43m            [49m[43mgcs_bucket[49m[38;5;241;43m=[39;49m[38;5;124;43m"[39;49m[38;5;124;43mpolicyengine-uk-data-private[39;49m[38;5;124;43m"[39;49m[43m,[49m
[1;32m     21[0m [43m        [49m[43m)[49m
[1;32m     22[0m         time_period [38;5;241m=[39m [38;5;28;01mNone[39;00m
[1;32m     23[0m     [38;5;28;01melif[39;00m country [38;5;241m==[39m [38;5;124m"[39m[38;5;124mus[39m[38;5;124m"[39m:

File [0;32m/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/policyengine/utils/data_download.py:48[0m, in [0;36mdownload[0;34m(filepath, huggingface_repo, gcs_bucket, huggingface_org)[0m
[1;32m     46[0m [38;5;28;01mif[39;00m data_file[38;5;241m.[39mgcs_bucket [38;5;129;01mis[39;00m [38;5;129;01mnot[39;00m [38;5;28;01mNone[39;00m:
[1;32m     47[0m     logging[38;5;241m.[39minfo([38;5;124m"[39m[38;5;124mUsing Google Cloud Storage for download.[39m[38;5;124m"[39m)
[0;32m---> 48[0m     [43mdownload_file_from_gcs[49m[43m([49m
[1;32m     49[0m [43m        [49m[43mbucket_name[49m[38;5;241;43m=[39;49m[43mdata_file[49m[38;5;241;43m.[39;49m[43mgcs_bucket[49m[43m,[49m
[1;32m     50[0m [43m        [49m[43mfile_name[49m[38;5;241;43m=[39;49m[43mfilepath[49m[43m,[49m
[1;32m     51[0m [43m        [49m[43mdestination_path[49m[38;5;241;43m=[39;49m[43mfilepath[49m[43m,[49m
[1;32m     52[0m [43m    [49m[43m)[49m
[1;32m     53[0m     [38;5;28;01mreturn[39;00m filepath
[1;32m     55[0m [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m(
[1;32m     56[0m     [38;5;124m"[39m[38;5;124mNo valid download method specified. Please provide either a Hugging Face repo or a Google Cloud Storage bucket.[39m[38;5;124m"[39m
[1;32m     57[0m )

File [0;32m/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/policyengine/utils/google_cloud_bucket.py:18[0m, in [0;36mdownload_file_from_gcs[0;34m(bucket_name, file_name, destination_path)[0m
[1;32m     15[0m [38;5;28;01mfrom[39;00m[38;5;250m [39m[38;5;21;01mgoogle[39;00m[38;5;21;01m.[39;00m[38;5;21;01mcloud[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m storage
[1;32m     17[0m [38;5;66;03m# Initialize a client[39;00m
[0;32m---> 18[0m client [38;5;241m=[39m [43mstorage[49m[38;5;241;43m.[39;49m[43mClient[49m[43m([49m[43m)[49m
[1;32m     20[0m [38;5;66;03m# Get the bucket[39;00m
[1;32m     21[0m bucket [38;5;241m=[39m client[38;5;241m.[39mbucket(bucket_name)

File [0;32m/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/google/cloud/storage/client.py:247[0m, in [0;36mClient.__init__[0;34m(self, project, credentials, _http, client_info, client_options, use_auth_w_custom_endpoint, extra_headers, api_key)[0m
[1;32m    244[0m             no_project [38;5;241m=[39m [38;5;28;01mTrue[39;00m
[1;32m    245[0m             project [38;5;241m=[39m [38;5;124m"[39m[38;5;124m<none>[39m[38;5;124m"[39m
[0;32m--> 247[0m [38;5;28;43msuper[39;49m[43m([49m[43mClient[49m[43m,[49m[43m [49m[38;5;28;43mself[39;49m[43m)[49m[38;5;241;43m.[39;49m[38;5;21;43m__init__[39;49m[43m([49m
[1;32m    248[0m [43m    [49m[43mproject[49m[38;5;241;43m=[39;49m[43mproject[49m[43m,[49m
[1;32m    249[0m [43m    [49m[43mcredentials[49m[38;5;241;43m=[39;49m[43mcredentials[49m[43m,[49m
[1;32m    250[0m [43m    [49m[43mclient_options[49m[38;5;241;43m=[39;49m[43mclient_options[49m[43m,[49m
[1;32m    251[0m [43m    [49m[43m_http[49m[38;5;241;43m=[39;49m[43m_http[49m[43m,[49m
[1;32m    252[0m [43m[49m[43m)[49m
[1;32m    254[0m [38;5;66;03m# Validate that the universe domain of the credentials matches the[39;00m
[1;32m    255[0m [38;5;66;03m# universe domain of the client.[39;00m
[1;32m    256[0m [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39m_credentials[38;5;241m.[39muniverse_domain [38;5;241m!=[39m [38;5;28mself[39m[38;5;241m.[39muniverse_domain:

File [0;32m/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/google/cloud/client/__init__.py:338[0m, in [0;36mClientWithProject.__init__[0;34m(self, project, credentials, client_options, _http)[0m
[1;32m    337[0m [38;5;28;01mdef[39;00m[38;5;250m [39m[38;5;21m__init__[39m([38;5;28mself[39m, project[38;5;241m=[39m[38;5;28;01mNone[39;00m, credentials[38;5;241m=[39m[38;5;28;01mNone[39;00m, client_options[38;5;241m=[39m[38;5;28;01mNone[39;00m, _http[38;5;241m=[39m[38;5;28;01mNone[39;00m):
[0;32m--> 338[0m     [43m_ClientProjectMixin[49m[38;5;241;43m.[39;49m[38;5;21;43m__init__[39;49m[43m([49m[38;5;28;43mself[39;49m[43m,[49m[43m [49m[43mproject[49m[38;5;241;43m=[39;49m[43mproject[49m[43m,[49m[43m [49m[43mcredentials[49m[38;5;241;43m=[39;49m[43mcredentials[49m[43m)[49m
[1;32m    339[0m     Client[38;5;241m.[39m[38;5;21m__init__[39m(
[1;32m    340[0m         [38;5;28mself[39m, credentials[38;5;241m=[39mcredentials, client_options[38;5;241m=[39mclient_options, _http[38;5;241m=[39m_http
[1;32m    341[0m     )

File [0;32m/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/google/cloud/client/__init__.py:286[0m, in [0;36m_ClientProjectMixin.__init__[0;34m(self, project, credentials)[0m
[1;32m    283[0m     project [38;5;241m=[39m [38;5;28mgetattr[39m(credentials, [38;5;124m"[39m[38;5;124mproject_id[39m[38;5;124m"[39m, [38;5;28;01mNone[39;00m)
[1;32m    285[0m [38;5;28;01mif[39;00m project [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m:
[0;32m--> 286[0m     project [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_determine_default[49m[43m([49m[43mproject[49m[43m)[49m
[1;32m    288[0m [38;5;28;01mif[39;00m project [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m:
[1;32m    289[0m     [38;5;28;01mraise[39;00m [38;5;167;01mEnvironmentError[39;00m(
[1;32m    290[0m         [38;5;124m"[39m[38;5;124mProject was not passed and could not be [39m[38;5;124m"[39m
[1;32m    291[0m         [38;5;124m"[39m[38;5;124mdetermined from the environment.[39m[38;5;124m"[39m
[1;32m    292[0m     )

File [0;32m/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/google/cloud/client/__init__.py:305[0m, in [0;36m_ClientProjectMixin._determine_default[0;34m(project)[0m
[1;32m    302[0m [38;5;129m@staticmethod[39m
[1;32m    303[0m [38;5;28;01mdef[39;00m[38;5;250m [39m[38;5;21m_determine_default[39m(project):
[1;32m    304[0m [38;5;250m    [39m[38;5;124;03m"""Helper:  use default project detection."""[39;00m
[0;32m--> 305[0m     [38;5;28;01mreturn[39;00m [43m_determine_default_project[49m[43m([49m[43mproject[49m[43m)[49m

File [0;32m/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/google/cloud/_helpers/__init__.py:152[0m, in [0;36m_determine_default_project[0;34m(project)[0m
[1;32m    140[0m [38;5;250m[39m[38;5;124;03m"""Determine default project ID explicitly or implicitly as fall-back.[39;00m
[1;32m    141[0m 
[1;32m    142[0m [38;5;124;03mSee :func:`google.auth.default` for details on how the default project[39;00m
[0;32m   (...)[0m
[1;32m    149[0m [38;5;124;03m:returns: Default project if it can be determined.[39;00m
[1;32m    150[0m [38;5;124;03m"""[39;00m
[1;32m    151[0m [38;5;28;01mif[39;00m project [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m:
[0;32m--> 152[0m     _, project [38;5;241m=[39m [43mgoogle[49m[38;5;241;43m.[39;49m[43mauth[49m[38;5;241;43m.[39;49m[43mdefault[49m[43m([49m[43m)[49m
[1;32m    153[0m [38;5;28;01mreturn[39;00m project

File [0;32m/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/google/auth/_default.py:685[0m, in [0;36mdefault[0;34m(scopes, request, quota_project_id, default_scopes)[0m
[1;32m    677[0m             _LOGGER[38;5;241m.[39mwarning(
[1;32m    678[0m                 [38;5;124m"[39m[38;5;124mNo project ID could be determined. Consider running [39m[38;5;124m"[39m
[1;32m    679[0m                 [38;5;124m"[39m[38;5;124m`gcloud config set project` or setting the [39m[38;5;132;01m%s[39;00m[38;5;124m [39m[38;5;124m"[39m
[1;32m    680[0m                 [38;5;124m"[39m[38;5;124menvironment variable[39m[38;5;124m"[39m,
[1;32m    681[0m                 environment_vars[38;5;241m.[39mPROJECT,
[1;32m    682[0m             )
[1;32m    683[0m         [38;5;28;01mreturn[39;00m credentials, effective_project_id
[0;32m--> 685[0m [38;5;28;01mraise[39;00m exceptions[38;5;241m.[39mDefaultCredentialsError(_CLOUD_SDK_MISSING_CREDENTIALS)

[0;31mDefaultCredentialsError[0m: Your default credentials were not found. To set up Application Default Credentials, see https://cloud.google.com/docs/authentication/external/set-up-adc for more information.

